sklearn/impute/_knn.py

```diff
@@ -178,14 +178,20 @@ def fit(self, X, y=None):
             raise ValueError(
                 "Expected n_neighbors > 0. Got {}".format(self.n_neighbors))
 
+        X = check_array(X, accept_sparse=False, dtype=FLOAT_DTYPES,
+                        force_all_finite=force_all_finite, copy=self.copy)
+
         X = self._validate_data(X, accept_sparse=False, dtype=FLOAT_DTYPES,
                                 force_all_finite=force_all_finite,
                                 copy=self.copy)
-        super()._fit_indicator(X)
 
         _check_weights(self.weights)
         self._fit_X = X
         self._mask_fit_X = _get_mask(self._fit_X, self.missing_values)
+
+        missing_mask = self._mask_fit_X
```

Comment:
please avoid this unnecessary intermediate variable

Reply:
Sure, I'll remove it at once